ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -127,6 +127,23 @@
 
   @Override
   public Future<Long> write(ByteBuffer src, Callback<Long> callback) {
```

Comment:
i would recommend doing the thing that was done with in the default impl in AsyncWritableChannel where a second future is created and completed from the callback. This will ensure that the future returned is completed after the buffer position is changed.